x-pack/plugin/sql/qa/server/single-node/src/test/java/org/elasticsearch/xpack/sql/qa/single_node/ConsistentFunctionArgHandlingIT.java

```diff
@@ -193,18 +194,17 @@ public void test() throws Exception {
                 final String functionCall = functionName + "(" + join(", ", functionCallArgs) + ")";
                 final String query = "SELECT " + functionCall + " FROM " + indexName + " WHERE docId = '" + testDocId + "'";
                 ResultSet retVal = esJdbc().createStatement().executeQuery(query);
-                
+
                 assertTrue(retVal.next());
                 results.add(tuple(functionName + "(" + join(", ", functionCallArgsForAssert) + ")", retVal.getObject(1)));
                 // only a single row should be returned
                 assertFalse(retVal.next());
 
-                Stream<Object> returnValueStream = results.stream().map(Tuple::v2);
```

Comment:
Why did you remove this and now go over the stream components twice?

Reply:
Sadly streams can only be iterated once, during the second iteration it gave `stream is already closed` exception.

Reply:
Why did the tests pass then?

Reply:
Because the second iteration only happens if the test finds an issue (to create the detailed message for error).